pkg/distributor/distributor.go

```diff
@@ -474,6 +497,17 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 		return &logproto.PushResponse{}, httpgrpc.Errorf(http.StatusUnprocessableEntity, validation.MissingStreamsErrorMsg)
 	}
 
+	if d.cfg.IngestLimitsEnabled {
+		exceedsLimits, err := d.exceedsLimits(ctx, tenantID, req.Streams)
+		if err != nil {
+			level.Error(d.logger).Log("msg", "failed to check if request exceeds limits, request has been accepted", "err", err)
+		} else if len(exceedsLimits.RejectedStreams) > 0 {
+			level.Error(d.logger).Log("msg", "request exceeded limits", "tenant", tenantID)
```

Comment:
should we return a proper error that can be understood by the client should be the same as ingester current error ?

Reply:
I would quite like to run it in ops first (in this dry run mode) before enabling it. Perhaps I could put it behind another feature flag?

Reply:
I'll merge this PR and we can discuss this offline 😄